packages/nextjs/src/server/index.ts

```diff
@@ -117,19 +113,67 @@ export function init(options: NodeOptions): void {
 
   nodeInit(opts);
 
-  const filterTransactions: EventProcessor = event => {
-    return event.type === 'transaction' && event.transaction === '/404' ? null : event;
+  const filterLowQualityTransactions: EventProcessor = event => {
+    if (event.type === 'transaction') {
+      // Filter out transactions for static assets
+      // This regex matches the default path to the static assets (`_next/static`) and could potentially filter out too many transactions.
+      // We match `/_next/static/` anywhere in the transaction name because its location may change with the basePath setting.
+      if (event.transaction?.match(/^GET (\/.*)?\/_next\/static\//)) {
+        return null;
+      }
+
+      // Filter out transactions for requests to the tunnel route
+      if (
+        globalWithInjectedValues.__sentryRewritesTunnelPath__ &&
+        event.transaction === `POST ${globalWithInjectedValues.__sentryRewritesTunnelPath__}`
+      ) {
+        return null;
+      }
+
+      // Filter out requests to resolve source maps for stack frames in dev mode
+      if (event.transaction?.match(/\/__nextjs_original-stack-frame/)) {
+        return null;
+      }
+
+      return event;
+    } else {
+      return event;
+    }
   };
+  filterLowQualityTransactions.id = 'NextLowQualityTransactionsFilter';
+  addEventProcessor(filterLowQualityTransactions);
+
+  const filterSentrySpans: EventProcessor = event => {
+    if (event.type === 'transaction') {
+      event.spans = event.spans?.filter(span => {
+        // Filter out spans for Sentry event sends
+        const httpTargetAttribute: unknown = span.data?.['http.target'];
+        if (typeof httpTargetAttribute === 'string') {
+          // TODO: Find a more robust matching logic
```

Comment:
We should use `suppressTracing` here 🤔 something to look into in the future!

Reply:
to be clear, with "here" I mean in the transport :D

Reply:
Yep, I'll add a TODO to potentially revisit this in the future.